File: amethyst.ui/src/types/OverlayConfiguration.ts
export type OverlayConfiguration = {
    scale: number;
    useBackground: boolean;
    backgroundColor: string;
    language: string;
}

export const DEFAULT_OVERLAY_CONFIGURATION: OverlayConfiguration = {
    scale: 1.0,
    useBackground: false,
    backgroundColor: '#00ff00',
    language: 'en',
}

File: jamster.engine/Configurations/OverlayConfiguration.cs
namespace jamster.Configurations;

public record OverlayConfiguration(float Scale, bool UseBackground, string BackgroundColor, string Language);

public sealed class OverlayConfigurationFactory : IConfigurationFactory<OverlayConfiguration>
{
    public OverlayConfiguration GetDefaultValue() => new(1.0f, false, "#00ff00", "en");
}